tools/cbmc/proofs/HTTP/global_state_HTTP.c

```diff
@@ -0,0 +1,201 @@
+  return ret;
+}
+
+IotNetworkError_t IotNetworkInterfaceCreate( void * pConnectionInfo,
+               void * pCredentialInfo,
+               void * pConnection ) {
+    *(char **)pConnection = malloc(1); /* network connection is opaque */
+    IotNetworkError_t error;
+    return error;
+}
+
+size_t IotNetworkInterfaceSend( void * pConnection,
+           const uint8_t * pMessage,
+           size_t messageLength ) {
+    size_t size;
+    __CPROVER_assume(size <= messageLength);
+    return size;
+}
+
+IotNetworkError_t IotNetworkInterfaceClose( void * pConnection ) {
+    IotNetworkError_t error;
+    return error;
+}
+
+size_t IotNetworkInterfaceReceive( void * pConnection,
+        uint8_t * pBuffer,
+        size_t bytesRequested ) {
+    uint8_t byte;
+    __CPROVER_array_copy(pBuffer,&byte);
+
+    size_t size;
+    __CPROVER_assume(size <= bytesRequested);
+    return size;
+}
+
+IotNetworkError_t IotNetworkInterfaceCallback( void * pConnection,
+                 IotNetworkReceiveCallback_t
+                 receiveCallback,
+                 void * pContext ) {
+    IotNetworkError_t error;
+    return error;
+}
+
+IotNetworkError_t IotNetworkInterfaceDestroy( void * pConnection ) {
+  IotNetworkError_t error;
+  return error;
+}
+
+IotNetworkInterface_t IOTNI = {
+  .create = IotNetworkInterfaceCreate,
+  .close = IotNetworkInterfaceClose,
+  .send = IotNetworkInterfaceSend,
+  .receive = IotNetworkInterfaceReceive,
+  .setReceiveCallback = IotNetworkInterfaceCallback,
+  .destroy = IotNetworkInterfaceDestroy
+};
+
+/* Models the Network Interface. */
+IotNetworkInterface_t *newNetworkInterface() {
+  return &IOTNI;
+}
+
+/* Creates a Connection Info and assigns memory accordingly. */
+IotHttpsConnectionInfo_t * newConnectionInfo() {
+  IotHttpsConnectionInfo_t * pConnInfo = safeMalloc(sizeof(IotHttpsConnectionInfo_t));
+  if(pConnInfo) {
+    pConnInfo->pNetworkInterface = newNetworkInterface();
+    /* The +1 is required for coverage */
+    __CPROVER_assume(pConnInfo->addressLen >= 0 && pConnInfo->addressLen <= IOT_HTTPS_MAX_HOST_NAME_LENGTH + 1);
+    pConnInfo->pAddress = safeMalloc(pConnInfo->addressLen);
+    /* The +1 is required for coverage */
+    __CPROVER_assume(pConnInfo->alpnProtocolsLen >= 0 && pConnInfo->alpnProtocolsLen <= IOT_HTTPS_MAX_ALPN_PROTOCOLS_LENGTH + 1);
+    pConnInfo->pAlpnProtocols = safeMalloc(pConnInfo->alpnProtocolsLen);
+    pConnInfo->pCaCert = malloc(sizeof(uint32_t));
+    pConnInfo->pClientCert = malloc(sizeof(uint32_t));
+    pConnInfo->pPrivateKey = malloc(sizeof(uint32_t));
+    pConnInfo->userBuffer.pBuffer = safeMalloc(sizeof(_connHandle_t));
+  }
+  return pConnInfo;
+}
+
+/* Creates a Connection Handle and assigns memory accordingly. */
+IotHttpsConnectionHandle_t newIotConnectionHandle () {
+  IotHttpsConnectionHandle_t pConnectionHandle = safeMalloc(sizeof(_connHandle_t));
+  if(pConnectionHandle) {
+    pConnectionHandle->pNetworkConnection = safeMalloc(sizeof(_connHandle_t));
```

Comment:
This is fine, pNetworkConnection is an opaque handle not part of this HTTPS Client library and is not used in the proof mocked functions, so it can be of any size. 
Maybe we could make it smaller and save memory (malloc(sizeof(uint32_t)).

Reply:
Ok. I will make it smaller.